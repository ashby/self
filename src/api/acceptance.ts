import config from '../config';
const { HIGHER_POWER_PATH } = config;

export const getAcceptance = async path => 
    fetch( `${HIGHER_POWER_PATH}/${path}/acceptance` );
export const postAcceptance = async ( path, acceptance ) => 
    fetch( `${HIGHER_POWER_PATH}/${path}/acceptance`, { method: 'post', body: acceptance } );
export const putAcceptance = async ( path, acceptance ) => 
    fetch( `${HIGHER_POWER_PATH}/${path}/acceptance`, { method: 'put', body: acceptance } );
export const deleteAcceptance = async path => fetch( `${HIGHER_POWER_PATH}/${path}/acceptance`, { method: 'delete' } );