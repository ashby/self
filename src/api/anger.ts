import config from '../config';
const { HIGHER_POWER_PATH } = config;

export const getAnger = async path => fetch( `${HIGHER_POWER_PATH}/${path}/anger` );
export const postAnger = async ( path, anger ) => fetch( `${HIGHER_POWER_PATH}/${path}/anger`, { method: 'post', body: anger } );
export const putAnger = async ( path, anger ) => fetch( `${HIGHER_POWER_PATH}/${path}/anger`, { method: 'put', body: anger } );
export const deleteAnger = async path => fetch( `${HIGHER_POWER_PATH}/${path}/anger`, { method: 'delete' } );
