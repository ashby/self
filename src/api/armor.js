import { HIGHER_POWER_PATH } from '../config';

export const getArmor = async path => fetch( `${HIGHER_POWER_PATH}/${path}/armor` );
export const postArmor = async ( path, armor ) => fetch( `${HIGHER_POWER_PATH}/${path}/armor`, { method: 'post', body: armor } );
export const putArmor = async ( path, armor ) => fetch( `${HIGHER_POWER_PATH}/${path}/armor`, { method: 'put', body: armor } );
export const deleteArmor = async path => fetch( `${HIGHER_POWER_PATH}/${path}/armor`, { method: 'delete' } );