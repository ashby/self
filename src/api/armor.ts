import config from '../config';
const { HIGHER_POWER_PATH } = config;

export const getArmor = async path => fetch( `${HIGHER_POWER_PATH}/${path}/armor` );
export const postArmor = async ( path, armor ) => fetch( `${HIGHER_POWER_PATH}/${path}/armor`, { method: 'post', body: armor } );
export const putArmor = async ( path, armor ) => fetch( `${HIGHER_POWER_PATH}/${path}/armor`, { method: 'put', body: armor } );
export const deleteArmor = async path => fetch( `${HIGHER_POWER_PATH}/${path}/armor`, { method: 'delete' } );