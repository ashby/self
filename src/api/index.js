export const getAnger = async path => fetch( `https://self/${path}/anger` );
export const postAnger = async ( path, anger ) => fetch( `https://self/${path}/anger`, { method: 'post', body: anger } );
export const putAnger = async ( path, anger ) => fetch( `https://self/${path}/anger`, { method: 'put', body: anger } );
export const deleteAnger = async path => fetch( `https://self/${path}/anger`, { method: 'delete' } );
export const getSelfPity = async path => fetch( `https://self/${path}/selfpity` );
export const postSelfPity = ( path, selfpity ) => fetch( `https://self/${path}/selfpity`, { method: 'post', body: selfpity } );
export const putSelfPity = ( path, selfpity ) => fetch( `https://self/${path}/selfpity`, { method: 'put', body: selfpity } );
export const deleteSelfPity = async path => fetch( `https://self/${path}/selfpity`, { method: 'delete' } );