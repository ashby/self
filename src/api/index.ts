export * from './anger';
export * from './boundary';
export * from './armor';
export * from './selfPity';
export * from './vulnerability';
export * from './acceptance';