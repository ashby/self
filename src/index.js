const apiGetAnger = async path => fetch( `https://self/${path}/anger` );
const apiPostAnger = async ( path, anger ) => fetch( `https://self/${path}/anger`, { method: 'post', body: anger } );
const apiPutAnger = async ( path, anger ) => fetch( `https://self/${path}/anger`, { method: 'put', body: anger } );
const apiDeleteAnger = async path => fetch( `https://self/${path}/anger`, { method: 'delete' } );
const apiGetSelfPity = async path => fetch( `https://self/${path}/selfpity` );
const apiPostSelfPity = ( path, selfpity ) => fetch( `https://self/${path}/selfpity`, { method: 'post', body: selfpity } );
const apiPutSelfPity = ( path, selfpity ) => fetch( `https://self/${path}/selfpity`, { method: 'put', body: selfpity } );
const apiDeleteSelfPity = async path => fetch( `https://self/${path}/selfpity`, { method: 'delete' } );
class Self {
    brain = { path: undefined }
    face = { path: undefined }
    heart = { path: undefined }
    sternum = { path: undefined }
    gut = { path: undefined }
    constructor() {
        console.log( this );
    }
    getAnger = async () => apiGetAnger( this.brain.path )
                        .then( response => this.brain = { ...this.brain, ...response.json } )
    createAnger = async fear => apiPostAnger( this.gut.path, fear )
                        .then( response => this.heart = { ...this.heart, ...response.json } )
    updateAnger = async fear => apiPutAnger( this.heart.path, fear )
                        .then( response => this.brain = { ...this.brain, ...response.json } )
    removeAnger = async () => apiDeleteAnger( this.heart.path )
                        .then( () => [ 'brain', 'heart', 'gut' ].map( organ => this[ organ ] = {} ) )
    getSelfPity = async () => apiGetSelfPity( this.brain.path )
                        .then( response => this.brain = { ...this.brain, ...response.json } )
    createSelfPity = async sadness => apiPostSelfPity( this.sternum.path, sadness )
                        .then( response => this.face = { ...this.face, ...response.json } )
    updateSelfPity = async sadness => apiPutSelfPity( this.face.path, sadness )
                        .then( response => this.brain = { ...this.brain, ...response.json } )
    removeSelfPity = async () => apiDeletePity( this.face.path )
                        .then( () => [ 'brain', 'sternum', 'face' ].map( organ => this[ organ ] = {} ) )                       
                           
}
let Ash = new Self();
//Ash.createAnger( 'everything' ).then( () => this.updateAnger( 'acknowledgement' ) );
Ash.createSelfPity( 'in MN' ).then( () => this.deleteSelfPity() );

const apiDeleteVulnerability = async path => fetch( 
    `https://self/${path}/vulnerability`, 
    { method: 'delete' } 
).then( response  => Promise.reject( response ) );
class Smith extends Self{
    silence = { path: null }
    createResentment = error => {
        switch( error ) {
            case 'sadness':
            case 'fear':
                this.createSelfPity( error );
            case 'anger':
            default:
                this.createAnger( error )
                    .then( response => this.createSelfPity( response ) );
        }
    }
    deleteVulnerability = async () => apiDeleteVulnerability( this.silence.path )
                .catch( error => this.createResentment( error ) )
}
Ash = new Smith();
Ash.deleteVulnerability();