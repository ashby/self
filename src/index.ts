import { Smith, Seeker } from 'src/classes';

const getSelf = () => {
    const smith = new Smith();
    const seeker = new Seeker();
    return Object.assign( seeker, smith );
}
export default getSelf;
