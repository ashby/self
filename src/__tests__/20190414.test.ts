import getSelf from '../';
import * as api from 'src/api';
import { 
    expectEmptyPartsExcept,
    expectEmptyVirtuesExcept,
    expectEmptyShieldsExcept,
    expectEmptyConstructsExcept
} from 'app-test-utils';

const ARMOR = 'patiently waiting to hear';
const SELF_PITY = 'maybe I\'m too intense';
const ANGER = 'i deserve better';
const VULNERABILITY = 'i care and worry';

const BOUNDARY = 'doesn\'t affect me';

describe( '20190414 Self::', () => {
    let Ash;
    beforeEach( () => {
        Ash = getSelf();
    } );
    it( 'can assign smith to seeker and see parts, shields and virtues', () => {
        const parts = Ash.seeParts();
        const shields = Ash.seeShields();
        const virtues = Ash.seeVirtues();
        expect( Object.keys( parts ) ).toHaveLength( 7 );
        expect( Object.keys( shields ) ).toHaveLength( 4 );
        expect( Object.keys( virtues ) ).toHaveLength( 3 );
        expect.assertions( 3 );
    } );
    it( 'can create Guard, manage silence, remove anger and self pity, and be ready with some vulnerability', async () => {
        spyOn( api, 'postBoundary' ).and.returnValue( Promise.resolve( [ BOUNDARY ] ) );
        spyOn( api, 'postArmor' ).and.returnValue( Promise.resolve( [ ARMOR ] ) );
        spyOn( api, 'putArmor' ).and.returnValue( Promise.resolve( [ ARMOR ] ) );
        spyOn( api, 'postAnger' ).and.returnValue( Promise.resolve( [ ANGER ] ) );
        spyOn( api, 'postSelfPity' ).and.returnValue( Promise.resolve( [ SELF_PITY ] ) );
        spyOn( api, 'deleteAnger' ).and.returnValue( Promise.resolve() );
        spyOn( api, 'deleteSelfPity' ).and.returnValue( Promise.resolve() );
        spyOn( api, 'putAnger' ).and.returnValue( Promise.resolve( [ BOUNDARY ] ) );
        spyOn( api, 'putSelfPity' ).and.returnValue( Promise.resolve( [ BOUNDARY ] ) );
        spyOn( api, 'getVulnerability' ).and.returnValue( Promise.resolve( [ VULNERABILITY ] ) );
        spyOn( api, 'deleteVulnerability' ).and.returnValue( Promise.reject( [ VULNERABILITY ] ) );

        await Ash.createAnger( ANGER );
        await Ash.createSelfPity( SELF_PITY );
        await Ash.createArmor( ARMOR );
        await Ash.removeVulnerability();

        expect( Guard.gate.boundary ).toEqual( [ BOUNDARY ] );
        expect( Ash.gut.anger ).toEqual( [ ANGER ] );
        expect( Ash.silence.armor ).toEqual( [ ARMOR ] );
        expect( Guard.face.selfPity ).toEqual( [ BOUNDARY ] );
        expect( Guard.mouth.anger ).toEqual( [ BOUNDARY ] );
        expect( Guard.denial.armor ).toEqual( [ ARMOR ] );
        expect( Guard.confusion.armor ).toEqual( [ ARMOR ] );

        const shields = Ash.seeShields();
        const shieldsCount = Object.keys( shields ).length -1;
        expectEmptyShieldsExcept( [ 'silence' ], shields );
        const guardShields = Guard.seeShields();
        const guardShieldsCount = Object.keys( guardShields ).length - 2;
        expectEmptyShieldsExcept( [ 'denial', 'confusion' ], guardShields );
        const guardConstructs = Guard.seeConstructs();
        const guardConstructsCount = Object.keys( guardConstructs ).length - 1;
        expectEmptyConstructsExcept( [ 'gate' ], guardConstructs )

        await Ash.getVulnerability();
        expect( Ash.mind.vulnerability ).toEqual( [ VULNERABILITY ] );
        const virtues = Ash.seeVirtues();
        const virtuesCount = Object.keys( virtues ).length - 1;
        expectEmptyVirtuesExcept( [ 'mind' ], virtues );

        await Ash.removeAnger();
        await Ash.removeSelfPity();
        await Guard.removeAnger();
        await Guard.removeSelfPity();
        const parts = Ash.seeParts();
        const partsCount = Object.keys( parts ).length;
        expectEmptyPartsExcept( [], parts );
        const guardParts = Guard.seeParts();
        const guartsPartsCount = Object.keys( guardParts ).length;
        expectEmptyPartsExcept( [], guardParts );

        expect( { parts,shields, virtues, guardParts, guardShields, guardConstructs } ).toMatchSnapshot();
        expect.assertions( 9 + virtuesCount + partsCount + guartsPartsCount + guardShieldsCount + shieldsCount + guardConstructsCount );
    } );

    it( 'can dismiss guard and create acceptance', () => {
        expect( true ).toEqual( false );
    } );
} );