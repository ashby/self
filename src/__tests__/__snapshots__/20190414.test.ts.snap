// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`20190414 Self:: can create Guard, manage silence, remove anger and self pity, and be ready with some vulnerability 1`] = `
Object {
  "guardConstructs": Object {
    "gate": Object {
      "boundary": Array [
        "doesn't affect me",
      ],
    },
    "tower": Object {
      "boundary": Object {},
    },
    "wall": Object {
      "boundary": Object {},
    },
  },
  "guardParts": Object {
    "brain": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "face": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "gut": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "heart": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "mouth": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "skin": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "sternum": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
  },
  "guardShields": Object {
    "confusion": Object {
      "armor": Array [
        "patiently waiting to hear",
      ],
    },
    "denial": Object {
      "armor": Array [
        "patiently waiting to hear",
      ],
    },
    "sarcasm": Object {
      "armor": Array [],
    },
    "silence": Object {
      "armor": Array [],
    },
  },
  "parts": Object {
    "brain": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "face": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "gut": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "heart": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "mouth": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "skin": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
    "sternum": Object {
      "anger": Array [],
      "selfPity": Array [],
    },
  },
  "shields": Object {
    "confusion": Object {
      "armor": Array [],
    },
    "denial": Object {
      "armor": Array [],
    },
    "sarcasm": Object {
      "armor": Array [],
    },
    "silence": Object {
      "armor": Array [
        "patiently waiting to hear",
      ],
    },
  },
  "virtues": Object {
    "love": Object {
      "vulnerability": Array [],
    },
    "mind": Object {
      "vulnerability": Array [
        "i care and worry",
      ],
    },
    "soul": Object {
      "vulnerability": Array [],
    },
  },
}
`;
